### ISSUE TITLE
{willAttemptReconnect: true} when calling .disconnect on server

### ISSUE THREAD

**Issue by ac74394** [rails/rails#41574]
This is my first-ever bug report so please let me know how to help.

Standard Rails ActionCable setup...

### Expected behavior
When calling ActionCable.server.remote_connections.where(current_user: User.first).disconnect, the client should not attempt to automatically reconnect.

### Actual behavior
calling ActionCable.server.remote_connections.where(current_user: User.first).disconnect does disconnect the client...

But when viewing console on client, {willAttemptReconnect: true} appears (from the ActionCable js script) and the client is then reconnected to the disconnected channels.

### System configuration
**Rails version**: 5.2.4.5

**Ruby version**: 2.5.8p224

**Comment by georgeclaghorn** [comment]
Are you able to reproduce on latest Rails 6.1? 5.2 is several years old and [no longer receiving bug fixes](https://guides.rubyonrails.org/maintenance_policy.html).

**Comment by ac74394** [comment]
Thanks for the quick response!  I'll wire up a new Rails app and take a look...

**Comment by ac74394** [comment]
Hello - I just updated the app to 6.1.3.  Same problem.

From rails c:
irb(main):002:0> ActionCable.server.remote_connections.where(current_user: User.first).disconnect
  User Load (0.8ms)  SELECT "users".* FROM "users" ORDER BY "users"."id" ASC LIMIT $1  [["LIMIT", 1]]
[ActionCable] Broadcasting to action_cable/Z2lkOi8vYWxnb25pdXMvVXNlci8x: {:type=>"disconnect"}
=> 1
irb(main):003:0> 

And, in the browser console:
{willAttemptReconnect: true}

Finally, in the server log
Removing connection (Z2lkOi8vYWxnb25pdXMvVXNlci8x)
Finished "/cable/" [WebSocket] for 127.0.0.1 at 2021-02-28 19:37:06 -0500
Started GET "/cable" for 127.0.0.1 at 2021-02-28 19:37:27 -0500
Started GET "/cable/" [WebSocket] for 127.0.0.1 at 2021-02-28 19:37:27 -0500
Successfully upgraded to WebSocket (REQUEST_METHOD: GET, HTTP_CONNECTION: Upgrade, HTTP_UPGRADE: websocket)
  User Load (0.4ms)  SELECT "users".* FROM "users" WHERE "users"."id" = $1 LIMIT $2  [["id", 1], ["LIMIT", 1]]
  ↳ app/channels/application_cable/connection.rb:17:in `find_verified_user'
Registered connection (Z2lkOi8vYWxnb25pdXMvVXNlci8x)

**Comment by ac74394** [comment]
So, I got everything to work the way I think it should, using the following workaround (though, it doesn't seem logical that willAttemptReconnect should be set to true when a disconnect command is sent from the server; but it's entirely possible that I am wrong with my logic!)...

On the client side, in the channel_name.js script, I added the following to the disconnected function:

`		  disconnected: function() {
		    App.channelName.unsubscribe();
		  },`

That seems to stop the client from automatically trying to reconnect.

I wish I could be more helpful debugging the JS side but that's far beyond my abilities...lol!

Thanks for your willingness to help and have a great one!
